### ISSUE TITLE
EXC_BAD_ACCESS BSGAppHangDetector.m:126

### ISSUE THREAD

**Issue by cwalo** [bugsnag/bugsnag-cocoa#1120]
### Describe the bug
We're seeing exceptions from BSGAppHangDetector. I took a look through the release notes and didn't see any mentions, but this may already be resolved. Stack trace is included in details below the fold.

### Steps to reproduce
N/A

### Environment
* Bugsnag version: 6.8.4, 6.9.0, 6.9.1
* CocoaPods version: N/A
* Carthage version (if any): N/A
* iOS/tvOS/macOS version(s): 14.2
* Simulator or physical device: Device
* Xcode version: 12.4
* Swift version (if applicable):

### Example Repo 

- [ ] Create a minimal repository that can reproduce the issue
- [ ] Link to it here:

[https://app.bugsnag.com/lyft/ios-all-apps/errors/60a7b9911584740008a2771b?filters[event.since][0][type]=eq&filters[event.since][0][value]=7d&filters[metaData.app.name][0][type]=eq&filters[metaData.app.name][0][value]=Driver&filters[event.message][0][type]=ne&filters[event.message][0][value]=The%20app%20was%20terminated%20while%20unresponsive&pivot_tab=release.seen_in](https://app.bugsnag.com/lyft/ios-all-apps/errors/60a7b9911584740008a2771b?filters[event.since][0][type]=eq&filters[event.since][0][value]=7d&filters[metaData.app.name][0][type]=eq&filters[metaData.app.name][0][value]=Driver&filters[event.message][0][type]=ne&filters[event.message][0][value]=The%20app%20was%20terminated%20while%20unresponsive&pivot_tab=release.seen_in)

<details><summary>Error messages:</summary>

```
CrashReporter Key:  8e8158548a43b9ab966fec7bff92a6be991a396f
Hardware Model:     iPhone10,5
Process:            Driver
Identifier:         com.lyft.ios.driver
Version:            1003.14.3
Role:               Foreground
OS Version:         iOS 14.2
Exception Type:     EXC_BAD_ACCESS 
Exception Subtype:  KERN_INVALID_ADDRESS


EXC_BAD_ACCESS: Attempted to dereference null pointer.

0  libsystem_platform.dylib __platform_strcmp
1  Driver                   +[BugsnagStackframe stackframesWithBacktrace:length:] (BugsnagStackframe.m:111:13)
2  Driver                   -[BugsnagThread(Recording) initWithMachThread:backtraceAddresses:backtraceLength:errorReportingThread:index:] (BugsnagThread+Recording.m:188:29)
3  Driver                   +[BugsnagThread(Recording) allThreadsWithCurrentThreadBacktrace:] (BugsnagThread+Recording.m:100:28)
4  Driver                   +[BugsnagThread(Recording) allThreads:callStackReturnAddresses:] (BugsnagThread+Recording.m:61:16)
5  Driver                   __40-[BSGAppHangDetector startWithDelegate:]_block_invoke_2 (BSGAppHangDetector.m:126:35)
6  libdispatch.dylib        __dispatch_client_callout
7  libdispatch.dylib        __dispatch_continuation_pop$VARIANT$armv81
8  libdispatch.dylib        __dispatch_source_invoke$VARIANT$armv81
9  libdispatch.dylib        __dispatch_lane_serial_drain$VARIANT$armv81
10 libdispatch.dylib        __dispatch_lane_invoke$VARIANT$armv81
11 libdispatch.dylib        __dispatch_workloop_worker_thread
12 libsystem_pthread.dylib  __pthread_wqthread
```
</details>

**Comment by yousif-bugsnag** [comment]
Hi @cwalo, 

Thanks for reporting this. We are taking a look into this and will update you as soon as we know more.

**Comment by yousif-bugsnag** [comment]
Hi @cwalo,

We've been looking into this and after some initial investigation we suspect that there's some stack corruption occurring here. 

Are you also seeing this crash in Apple's crash reporter? If so, would you be able to share the Apple crash report with us? You can send it to us at support@bugsnag.com if you'd prefer.

**Comment by keitzer** [comment]
@yousif-bugsnag Apple's crash reporter, as in... the Xcode > Organizer > Crashes location? Or what are your referring to?

**Comment by nickdowell** [comment]
@keitzer that's right, [Crashes organizer in Xcode](https://developer.apple.com/documentation/xcode/acquiring-crash-reports-and-diagnostic-logs#Collect-Crash-Reports-from-TestFlight-and-the-App-Store) 👍

**Comment by keitzer** [comment]
Moved to email, issue 32116

**Comment by yousif-bugsnag** [comment]
This has now been addressed in [v6.12.0](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.12.0) (#1170). Thanks for raising this.

**Comment by sethfri** [comment]
I'm curious because I've had issues closed in the past that were just suspected to be memory corruption. How did this go from memory corruption to being an actual issue in Bugsnag?

**Comment by mattdyoung** [comment]
Hi @sethfri 

We were never able to definitively identify the cause of this crash as an issue with Bugsnag, but we implemented some fixes in this area in v6.10.1 and v6.12.0 for other issues we did identify.

We've been monitoring crashes in this area of code over time and have seen a decline as these newer versions were adopted, so we believe this crash is likely to have been resolved as a side-effect of our changes.

If you do see any significant numbers of these crashes from versions after v6.12.0 we'd be very interested in the details.
